feature/com-178 setting deploy when merging into main

SireeKoolenWijkstra opened the pull request:
- add deploy yml file so that tests, build and deploy will be run before release to main. 
- set in Settings -> pages -> Source to main

SireeKoolenWijkstra committed:
aad deploy yml file so that tests will be run before release to main

thomasm0 commented:
If you compare https://github.com/Amsterdam/ee-docs/tree/gh-pages to https://github.com/Amsterdam/ee-docs/tree/main you can see `gh-pages` is just the generated docusaurus static website files, whereas in `main` these files don't exist (currently we don't commit the build files).

https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

I think we need to remove `build` from `.gitignore` so these files are commited. Then set the repo's github pages to target `main` and the `build` directory.

SireeKoolenWijkstra committed:
Add deploy yml so that deploy will happen automatically when push to main happens

Source for changes: https://docusaurus.io/docs/deployment#triggering-deployment-with-github-actions

SireeKoolenWijkstra committed:
include build files so that automatic deploying actually has files to deploy from

thomasm0 reviewed (approved):
Looks good, let's try do a deploy tomorrow. I'm also searching for a better explanation of the codeql alert, as that may become quite annoying if it pops up everytime we want to do a release.
